niceman/resource/tests/test_ssh.py

```diff
@@ -17,10 +17,42 @@
 from ...utils import swallow_logs
 from ...tests.utils import assert_in, skip_if_no_network
 from ..base import ResourceManager
+from ...cmd import Runner
 
+import pytest
+from nose import SkipTest
 
-@skip_if_no_network
-def test_ssh_class():
+@pytest.fixture(scope='module')
+def setup_docker():
+    """pytest fixture for tests needing a running docker container
+
+    on setup, this fixture ensures that a docker container that maps 
+    host port 49000 to container port 22 is running and starts one if necessary
+
+    on teardown, this fixture stops the docker container if it was started by 
+    the fixture
+    """
+    skip_if_no_network()
+    stdout, _ = Runner().run(['docker', 'ps'])
+    if '0.0.0.0:49000->22/tcp' in stdout:
+        stop_container = False
+    else:
+        args = ['docker', 
+                'run', 
+                '-d', 
+                '--rm', 
+                '-p', 
+                '49000:22', 
```

Comment:
wouldn't it cause trouble for those who run tests locally and have sshd running already on port 22?  why not to just retain/reuse port 49000 or some other privileged, but unlikely to be used port (2223?)?

Reply:
I don't understand -- 22 is the container port and 49000 is the host port, so it doesn't matter if the host is running sshd.

Reply:
ah, I took it wrong direction probably